tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java

```diff
@@ -143,22 +312,134 @@ public void testListGroupCommand(String quorum) throws Exception {
+            out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs4);
+                return null;
+            }));
+            return out.get().contains("TYPE") && out.get().contains("STATE") && out.get().contains(simpleGroup) && out.get().contains(GROUP);
+        }, "Expected to find " + simpleGroup + ", " + GROUP + " and the header, but found " + out.get());
+
+        String[] cgcArgs5 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "Stable"};
+        TestUtils.waitForCondition(() -> {
+            out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs5);
+                return null;
+            }));
             return out.get().contains("STATE") && out.get().contains(GROUP) && out.get().contains("Stable");
         }, "Expected to find " + GROUP + " in state Stable and the header, but found " + out.get());
 
-        String[] cgcArgs4 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "stable"};
+        String[] cgcArgs6 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--state", "stable"};
         TestUtils.waitForCondition(() -> {
             out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
-                ConsumerGroupCommand.main(cgcArgs4);
+                ConsumerGroupCommand.main(cgcArgs6);
                 return null;
             }));
             return out.get().contains("STATE") && out.get().contains(GROUP) && out.get().contains("Stable");
         }, "Expected to find " + GROUP + " in state Stable and the header, but found " + out.get());
+
+        String[] cgcArgs7 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--type", "Classic"};
+        TestUtils.waitForCondition(() -> {
+            out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs7);
+                return null;
+            }));
+            return out.get().contains("TYPE") && out.get().contains("Classic") && !out.get().contains("STATE") &&
+                out.get().contains(simpleGroup) && out.get().contains(GROUP);
+        }, "Expected to find " + GROUP + " and the header, but found " + out.get());
+
+        String[] cgcArgs8 = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list", "--type", "classic"};
+        TestUtils.waitForCondition(() -> {
+            out.set(kafka.utils.TestUtils.grabConsoleOutput(() -> {
+                ConsumerGroupCommand.main(cgcArgs8);
+                return null;
+            }));
+            return out.get().contains("TYPE") && out.get().contains("Classic") && !out.get().contains("STATE") &&
+                out.get().contains(simpleGroup) && out.get().contains(GROUP);
+        }, "Expected to find " + GROUP + " and the header, but found " + out.get());
```

Comment:
I understand that you follow the pattern in place but I find all those tests really hard to read/follow. I took a stab at improving them: https://github.com/apache/kafka/pull/15382. Please let me know what you think.

Reply:
Thanks a lot! I'll take a look!